extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -334,64 +333,60 @@ public function getPage() {
 			$title = $this->getTitleFromRequest();
 			$data = [ ];
 
+			if ( !$title->isKnown() ) {
+				throw new NotFoundApiException( 'Page doesn\'t exist' );
+			}
+
 			// getPage is cached (see the bottom of the method body) so there is no need for additional caching here
 			$article = Article::newFromID( $title->getArticleId() );
-			$articleExists = $article instanceof Article;
 
-			if ( $articleExists && $title->isRedirect() ) {
+			if ( $title->isRedirect() ) {
 				list( $title, $article, $data ) = $this->handleRedirect( $title, $article, $data );
```

Comment:
Redirect to deleted page will result in correct api response with `redirectEmptyTarget` flag set. Is this behavior correct? Are we handling it on mercury side correctly?